roles/patroni/templates/patroni.yml.j2

```diff
@@ -35,7 +35,16 @@ etcd3:
 {% endif %}
 {% if dcs_exists|bool and dcs_type == 'etcd' %}
 etcd3:
-  hosts: {% for etcd_hosts in patroni_etcd_hosts %}{{etcd_hosts.host}}:{{etcd_hosts.port}}{% if not loop.last %},{% endif %}{% endfor %}
+  hosts: {% for etcd_hosts in patroni_etcd_hosts %}{{etcd_hosts.host}}:{{etcd_hosts.port}}{% if not loop.last %},{% endif %}{% endfor +%}
+  {% if patroni_etcd_username | default("") | length > 0 %}
+  username: {{ patroni_etcd_username | default("") }}
+  {% endif %}
+  {% if patroni_etcd_password | default("") | length > 0 %}
+  password: {{ patroni_etcd_password }}
+  {% endif %}
+  {% if patroni_etcd_protocol | default("") | length > 0 %}
```

Comment:
```suggestion
  {% if patroni_etcd_username | default('') | length > 0 %}
  username: {{ patroni_etcd_username | default('') }}
  {% endif %}
  {% if patroni_etcd_password | default('') | length > 0 %}
  password: {{ patroni_etcd_password }}
  {% endif %}
  {% if patroni_etcd_protocol | default('') | length > 0 %}
```

Reply:
In the provided template snippet, it's advisable to use single quotes (') instead of double quotes (") for string delimiters to ensure consistency and to avoid potential issues with character escaping. Single quotes are less likely to cause escaping issues as they don't interfere with embedded variables or special characters, making the code more readable and maintainable.